### PR TITLE
media-tv/plex-media-server: version bump to 1.15.0.659

### PR DESCRIPTION
**Pull request by ephemer0l**
Hi. I'd like to take this over as a proxy-maintainer.

Closes Bug: https://bugs.gentoo.org/661734
and 
https://github.com/gentoo/gentoo/pull/10304

### CODE REVIEW

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+LICENSE="Plex"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+RESTRICT="mirror bindist strip"
+IUSE="avahi system-openssl"
+
+RDEPEND="avahi? ( net-dns/avahi )
+	system-openssl? ( dev-libs/openssl:0 )"
+
+QA_DESKTOP_FILE="usr/share/applications/plexmediamanager.desktop"
```
Comment (gyakovlev) [inline media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: since you already patch it, is it possible to patch it so there is no violation?

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+QA_PREBUILT="*"
+QA_MULTILIB_PATHS=( "usr/lib/${_APPNAME}/.*" )
+
+S="${WORKDIR}"
+
+PATCHES=( "${FILESDIR}/plexmediamanager.desktop.patch" )
```
Comment (gyakovlev) [inline media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: I see no patch committed or on filesdir. forgot to include into commit?

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+DESCRIPTION="A free media library that is intended for use with a plex client"
+HOMEPAGE="https://www.plex.tv/"
+
+_APPNAME="plexmediaserver"
+_USERNAME="plex"
+_SHORTNAME="${_USERNAME}"
+_COMMIT="cc260c476"
+_FULL_VERSION="${PV}-${_COMMIT}"
+
```
[the remark this answers is not in the document]
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: Yes, this was picked up from what was in the tree, previously. I will clean it up so it's easier to read.

Thank you

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+URI="https://downloads.plex.tv/plex-media-server"
+SRC_URI="
+	amd64? ( ${URI}/${_FULL_VERSION}/${_APPNAME}_${_FULL_VERSION}_amd64.deb )
+	x86? ( ${URI}/${_FULL_VERSION}/${_APPNAME}_${_FULL_VERSION}_i386.deb )"
+
+LICENSE="Plex"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+RESTRICT="mirror bindist strip"
+IUSE="avahi system-openssl"
```
[the remark this answers is not in the document]
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: Indeed.

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+_SHORTNAME="${_USERNAME}"
+_COMMIT="cc260c476"
+_FULL_VERSION="${PV}-${_COMMIT}"
+
+URI="https://downloads.plex.tv/plex-media-server"
+SRC_URI="
+	amd64? ( ${URI}/${_FULL_VERSION}/${_APPNAME}_${_FULL_VERSION}_amd64.deb )
+	x86? ( ${URI}/${_FULL_VERSION}/${_APPNAME}_${_FULL_VERSION}_i386.deb )"
+
+LICENSE="Plex"
```
[the remark this answers is not in the document]
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: There is actually a Plex license in /usr/portage/licenses/Plex I have not reviewed if it's been updated upon release.

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	local config_vanilla="/etc/default/${_APPNAME}"
+	local config_path="/etc/${_SHORTNAME}"
+	dodir "${config_path}"
+	insinto "${config_path}"
+	doins "${config_vanilla#/}"
+
+	sed -e "s#${config_vanilla}#${config_path}/$(basename "${config_vanilla}")#g" \
+		-i "${S}"/usr/sbin/start_pms || die
+
+	rm -r "usr/share/doc" || die
```
[the remark this answers is not in the document]
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: Previous authors work. Would you like it behind a USE, or is USE=doc more for extra document generation?

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild (hunk 5 above)
[the remark this answers is not in the document]
Comment (gyakovlev) [reply media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: license_groups is another thing. sometimes it's needed to mark `Plex` license as EULA.

here's an example how to do that, basically add `Plex` to `EULA` line
https://github.com/gentoo/gentoo/commit/bfae0756d50861d9efe1661c678289f0f79ada85#diff-5d84adcc05fe18ca8c3e9162fecea1e9
you can add it as a separate commit to this branch.
if you are not comfortable with committing to profiles dir it I can add it on merge.

File: media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild (hunk 2 above)
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.14.1.5488.ebuild]: Included. Thanks.

**Comment by ephemer0l** [comment]
> does it even use avahi?
> if it does, probably
> 
> ```shell
> depend() {
>     use avahi-daemon
> }
> ```
> 
> is the way to go, it will not start it if it's not enabled by user, but will reorder itself to start after it if it's enabled.
> can't test this, unfortunately.

It's actually all binary, and I don't think it needed it in the first place. I'm trying to pick it apart to see what I can replace with our own versions of the deps from portage.

Currently, upstream is having issues with it's mirrors, I have some more changes to make to another version that should be released soon I'd like to include.

Thank you for your time.

File: media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+SRC_URI="
+	amd64? ( ${URI}/${MY_PV}/debian/plexmediaserver_${MY_PV}_amd64.deb )
+	x86? ( ${URI}/${MY_PV}/debian/plexmediaserver_${MY_PV}_i386.deb )"
+
+LICENSE="Plex"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+RESTRICT="mirror bindist strip"
+
+DEPEND="sys-apps/fix-gnustack"
```
Comment (gyakovlev) [inline media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild]: it's `BDEPEND` in `EAPI=7`

File: media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+QA_DESKTOP_FILE="usr/share/applications/plexmediamanager.desktop"
```
Comment (gyakovlev) [inline media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild]: does not seem to be needed.

File: profiles/license_groups [hunk 9]
```diff
@@ -71,7 +71,7 @@ BINARY-REDISTRIBUTABLE @FREE bh-luxi Broadcom Dina intel-ucode ipw2100-fw ipw220
...
+EULA AdobeFlash-11.x AMD-GPU-PRO-EULA AnyDesk-TOS ArxFatalis-EULA-JoWooD baudline BCS bf1942-lnxded CAPYBARA-EULA Coherent-Graphics CROSSOVER-2 DOOM3 ETQW f.lux FAH-EULA-2014 FraunhoferFDK GameFront Gameplay-Group-EULA geekbench genymotion GIMPS GOG-EULA google-chrome Google-TOS Intel-SDP Introversion LastPass LOKI-EULA LRCTF MakeMKV-EULA Mendeley-terms Microsemi Mojang MTA-0.5 NVIDIA-CODEC-SDK NVIDIA-CUDA OPERA-12 OPERA-2014 Oracle-BCLA-JavaSE PAPERS-PLEASE Plex POMPOM postal2 Primate-Plunge protonmail-bridge-EULA PUEL Q3AEULA Q3AEULA-20000111 QUAKE4 Quartus-prime-megacore RAR RTCW RTCW-ETEULA RuneScape-EULA SJ-Labs Sourcetrail SPS StarUML-EULA Steam supermicro teamspeak3 TeamViewer THINKTANKS TIK ubiquiti ut2003 ut2003-demo Vivaldi worklog-assistant zi-labone
```
Comment (gyakovlev) [inline profiles/license_groups]: this needs to be a separate commit. please split it out.

File: media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild (hunk 7 above)
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild]: Fixed, thanks.

File: media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild (hunk 8 above)
Comment (ephemer0l) [reply media-tv/plex-media-server/plex-media-server-1.15.0.659.ebuild]: Removed

**Comment by ephemer0l** [comment]
> Package installs a lot of python stuff, but it;s not reflected in the ebuild in any way.
> https://wiki.gentoo.org/wiki/Project:Python/python-single-r1
> 
> Also it seems to be bundling intel vaapi driver, not sure what to do with it,I don't have intel card/cpu to test.

It's all binary, I hesitate to untangle it because I've had issues with symbol lookup errors from various things when I was poking at it.

**Comment by ephemer0l** [comment]
I will change the status of the Plex license as EULA in another pull.
